### PR TITLE
feat: autoload helpers

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add functionality to autoload helpers

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: app/Config/Autoload.php [hunk 1]
```diff
@@ -31,12 +31,12 @@ class Autoload extends AutoloadConfig
      * else you will need to modify all of those classes for this to work.
      *
      * Prototype:
-     *```
+     * ```
```
Comment (MGatner) [inline app/Config/Autoload.php:34]: Let's just drop all of these. No need for markdown back-ticks in block comments in a PHP file 🤔
```suggestion
```

File: system/Autoloader/Autoloader.php [hunk 2]
```diff
@@ -71,10 +71,19 @@ class Autoloader
...
      */
     protected $files = [];
 
+    /**
+     * Stores helper list.
+     *
+     * @var string[]
+     * @phpstan-var list<string>
+     */
+    protected $helpers = [];
```
Comment (MGatner) [inline system/Autoloader/Autoloader.php:85]: Let's consume our own feature:
```suggestion
    protected $helpers = ['url'];
```
Then use `array_merge()` with the Config and drop the helper call from bootstrap.php

**Review by MGatner — CHANGES_REQUESTED** [review]
Impressive moves on the tests! I have a flow suggestion, and obviously docs, but the rest is looking good.

File: app/Config/Autoload.php (hunk 1 above)
Comment (kenjis) [reply app/Config/Autoload.php]: Done.

File: system/Autoloader/Autoloader.php (hunk 2 above)
Comment (kenjis) [reply system/Autoloader/Autoloader.php]: Done.

**Comment by kenjis** [comment]
Added the docs.

File: tests/system/Autoloader/AutoloaderTest.php [hunk 3]
```diff
@@ -284,29 +294,45 @@ public function testFindsComposerRoutesWithComposerPathNotFound()
...
+    }
+
+    public function testLoadHelpers()
+    {
+        $config            = new Autoload();
+        $config->helpers[] = 'form';
+
+        $loader = new Autoloader();
+        $loader->initialize($config, new Modules());
+
```
Comment (MGatner) [inline tests/system/Autoloader/AutoloaderTest.php:331]: Does this need to be `@runInSeparateProcess` to be definitive? One way to check:
```suggestion
        $this->assertFalse(function_exists('form_open'));

```

Comment (kenjis) [reply tests/system/Autoloader/AutoloaderTest.php:335]: Yes! Done.

**Review by MGatner — APPROVED** [review]
Thanks!

**Pull request merged**